src/components/modal/modal-end-chat.js

```diff
@@ -0,0 +1,81 @@
+import React, { Component } from 'react';
+import '../log-in/log-in.scss';
+import { Link } from "react-router-dom";
+
+class ModalEndChat extends Component {
+  state = {
+    feedback: 0,
+    // questionid: this.props.modalRef.questionid,
+    // expiration: Date.now() + 30
+  }
+
+  closesQuestion = () => {
+    const token = localStorage.getItem('token');
+    const body = {
+      "karma": this.state.feedback,
+      "credits": 50
+    }
+
+    fetch(`http://localhost:4000/questions/${this.props.questionId}/feedback`, {
+      method: 'PUT', 
+      headers: { 
+        'Authorization' : 'Bearer ' + token,
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(body)
+    })
+      .then(res => res.json())
+      .then(res=> console.log('question CLOSED', res))
+  }
+
+  chatFeedback = (e) => {
+    e.preventDefault();
+    // this.props.sendOffer(this.state) //Amber TTD: put feedback in here
+    this.props.closeChatModal()
+    this.closesQuestion();
+    this.props.history.push('/');
+  }
+
+  setFeedback = (e, num) => {
+    e.preventDefault();
+    this.setState({feedback: num})
+    console.log(e, this.state.feedback)
```

Comment:
If I click 0 karma and then click 1 karma, I am console logging the previous number instead of the current number.